examples/rp2040.rs

```diff
@@ -0,0 +1,73 @@
+#![no_std]
+#![no_main]
+#![feature(type_alias_impl_trait)]
+
+use panic_probe as _;
+use defmt_rtt as _;
+
+use core::cell::RefCell;
+
+use nrf24l01::{TXConfig, PALevel, NRF24L01, OperatingMode};
+use embassy_embedded_hal::shared_bus::blocking::spi::SpiDeviceWithConfig;
+use embassy_sync::blocking_mutex::{Mutex, raw::NoopRawMutex};
+use embassy_executor::Spawner;
+use embassy_rp::{
+    spi::Spi,
+    gpio::{Level, Output, Pin}, spi::Blocking
+};
+use embassy_time::{Duration, Timer};
+
+async fn blink<T>(led: &mut Output<'_, T>) where T: Pin {
+    led.set_high();
+    Timer::after(Duration::from_millis(100)).await;
+    led.set_low();
+    Timer::after(Duration::from_millis(100)).await;
+}
+
+#[embassy_executor::main]
+async fn main(_spawner: Spawner) {
+    // peripherals
+    let peripherals_config = embassy_rp::config::Config::default();
+    let peripherals = embassy_rp::init(peripherals_config);
+    // LED
+    let mut led_output = Output::new(peripherals.PIN_25, Level::Low);
+    // SPI
+    let mut spi_config = embassy_rp::spi::Config::default();
+    spi_config.frequency = 10_000_000; // TODO: is this right?
+    spi_config.phase = embassy_rp::spi::Phase::CaptureOnSecondTransition; // TODO: is this right?
+    spi_config.polarity = embassy_rp::spi::Polarity::IdleHigh; // TODO: is this right?
+    let clk = peripherals.PIN_10;
+    let mosi = peripherals.PIN_11;
+    let miso =  peripherals.PIN_12;
+    let cs = peripherals.PIN_9;
+    let cs_output = Output::new(cs, Level::High);
+    let spi: Spi<'_, _, Blocking> = Spi::new_blocking(peripherals.SPI1, clk, mosi, miso, spi_config.clone());
+    let spi_bus: Mutex<NoopRawMutex, _> = Mutex::new(RefCell::new(spi));
+    let spi_device = SpiDeviceWithConfig::new(&spi_bus, cs_output, spi_config);
+    // NRF24L01
+    let nrf24l01_config = TXConfig {
+        channel: 108,
+        pa_level: PALevel::Low,
+        pipe0_address: *b"abcde",
+        max_retries: 3,
+        retry_delay: 2,
+        ..Default::default()
+    };
+    let mut device = NRF24L01::new(spi_device, cs_output).unwrap(); // TODO: how to share cs_output with SpiDeviceWithConfig because it gets moved?
```

Comment:
@henrikberg any recommendations on this one?

```
error[E0382]: use of moved value: `cs_output`
  --> examples/rp2040.rs:57:48
   |
44 |     let cs_output = Output::new(cs, Level::High);
   |         --------- move occurs because `cs_output` has type `Output<'_, PIN_9>`, which does not implement the `Copy` trait
...
47 |     let spi_device = SpiDeviceWithConfig::new(&spi_bus, cs_output, spi_config);
   |                                                         --------- value moved here
...
57 |     let mut device = NRF24L01::new(spi_device, cs_output).unwrap(); // TODO: how to share cs_output with SpiDeviceWithConfig because it gets moved?
   |                                                ^^^^^^^^^ value used here after move
```

I don't see how it is currently possible to share what's effectively the `CS` output pin with both `SpiDeviceWithConfig` and then this library `NRF24L01`

Reply:
I just realized `CE` pin isn't `CS` pin.